### DISCUSSION TITLE
Jdeps and Quarkus

### DISCUSSION THREAD

**Discussion by uPagge** [quarkusio/quarkus#30752] [Q&A]
I'm trying to use Jlink and Jdeps to reduce the size of the docker image. And I faced an unpleasant feature.

If I build an application using gradle in fast-jar, I get the following when calling the jdeps command

```sh
jdeps --multi-release 17 --ignore-missing-deps --recursive  --print-module-deps --class-path "build/quarkus-app/lib/boot/*:build/quarkus-app/lib/main/*:build/quarkus-app/quarkus/*:build/quarkus-app/*" build/quarkus-app/app/*

java.base,java.desktop,java.naming,java.rmi,java.sql,jdk.compiler,jdk.unsupported
```

If I build the same application using uber-jar, I'll get the following command output

```sh
jdeps --multi-release 17 --ignore-missing-deps --recursive --print-module-deps build/*.jar

java.base,java.desktop,java.management,java.naming,java.rmi,java.security.jgss,java.security.sasl,java.sql,jdk.compiler,jdk.unsupported
```

As you can see, there are many more modules and I can't understand what I'm missing. And the second conclusion is correct, but the first one is not, since my application crashes at startup when using the first option.

**Comment by uPagge** [comment]
And the jdk.zipfs module for some reason was not defined by any of the methods. Although it is needed, without it it crashes at startup

**Reply by fwelland** [reply]
Similar experience (missing jdk.zipfs)   but also missing  jdk.crypto.ec     

Maybe because of  quarkus-oidc?

At any rate, is this because: 

       of something wrong with jdeps? , 
       how one uses jdeps against an quarkus app?   or
       that module meta data in quarkus libs aren't quite right?
